packages/react/src/components/Popover/index.test.tsx

```diff
@@ -0,0 +1,311 @@
+import React, { useRef } from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import axe from '../../axe';
+import Popover from '../Popover';
+import AriaIsolate from '../../utils/aria-isolate';
+
+import type { PopoverProps } from '../Popover';
+
+let wrapperNode: HTMLDivElement | null;
+beforeEach(() => {
+  wrapperNode = document.createElement('div');
+  wrapperNode.innerHTML = `
+    <button data-test>Click Me!</button>
+    <div></div>
+  `;
+  document.body.appendChild(wrapperNode);
+});
+
+afterEach(() => {
+  document.body.innerHTML = '';
+  wrapperNode = null;
+});
+
+const Wrapper = ({
+  buttonProps = {},
+  tooltipProps = {}
+}: {
+  buttonProps?: React.HTMLAttributes<HTMLButtonElement>;
+  tooltipProps?: Omit<Partial<PopoverProps>, 'variant'>;
+}) => {
+  const ref = useRef<HTMLButtonElement>(null);
+  const onClose = jest.fn();
+
+  return (
+    <React.Fragment>
+      <button ref={ref} {...buttonProps}>
+        button
+      </button>
+      <Popover
+        variant="custom"
+        aria-labelledby="test-popover-title"
```

Comment:
may I ask you to clarify more? I am not sure that I understand correctly

Reply:
This `aria-labelledby` appears to be pointing to a label that doesn't exist which would be an accessibility issue.

Additionally, we need to be sure components have a test for each significant variant/state, example below: https://github.com/dequelabs/cauldron/blob/d122f709347cbf3de34d36d83e287af6e60b1d91/packages/react/src/components/Checkbox/Checkbox.test.tsx#L165-L187

Reply:
Added a case when we provide `aria-label`, and popover will not have `aria-labelledby` attr

Reply:
It's still not clear why this is set by default along with other some of the other properties. The "default" props should be limited to properties that would be commonly set across all testing states, so things like `show` or `target`. 

To be clear, we should not be setting `aria-labelledby` or `variant` in the default wrapper.

Reply:
Per the original comment: https://github.com/dequelabs/cauldron/pull/1505/files#r1635340561

`aria-labelledby` should be an id ref pointed to an element that exists. While I understand that this is a "custom" variant, we should be writing the test cases in ways that are using the label correctly. If you don't want to create the element, you can instead use "aria-label" for these instances.

The other reason not to specify one of the label attributes as a default property, is that it makes it difficult to test `aria-label` and `aria-labelledby` independently because there's already a default value set. When both are set, `aria-label` takes precedent meaning that we wouldn't be able to catch accessibility issues with `aria-labelledby` because the name is coming from `aria-label`.

Reply:
@scurker 
Orest helper me with this. Since we scanned `container` with `axe` -> I did not see any accessibility errors (we need to use `baseElement` instead of `container` for portals) - https://github.com/nickcolley/jest-axe?tab=readme-ov-file#testing-react-with-react-testing-library

1. Added a correct element with `id` on which popover would be `labelledby` for `<Wrapper />`
2. Fixed tests for axe violations (added separate test for variant `custom`, for aria-labelledby and aria-label)